### PR TITLE
[Oneshot IPA] fix test to honor QUERY_SIZE

### PR DESCRIPTION
**Pull request by richajaindce**
remove NUM_USERS requirement from oneshot so that QUERY_SIZE is honored while generating input for the test

### CODE REVIEW

File: benches/oneshot/ipa.rs [hunk 1]
```diff
@@ -28,21 +27,20 @@ async fn main() -> Result<(), Error> {
     let mut total_count = 0;
 
     let mut random_user_records = Vec::with_capacity(NUM_USERS);
```
Comment (martinthomson) [inline benches/oneshot/ipa.rs:29]: This is your only use for NUM_USERS now.

I would suggest that you calculate this value.  Setting it to QUERY_SIZE / Expected(records_per_user) is probably better than keeping a constant that will rot.

Also, I think we should change the distribution from which we draw the number of records for each user.  Uniform is not that realistic.

Comment (richajaindce) [reply benches/oneshot/ipa.rs:29]: we actually don't draw from a fixed number of users. Inside the function, it generates a random number upto MAX_RECORDS_PER_USER and uses that to decide the count.

File: benches/oneshot/ipa.rs [hunk 2]
```diff
@@ -13,36 +13,33 @@ use std::time::Instant;
...
             &mut rng,
             MAX_RECORDS_PER_USER,
             MAX_BREAKDOWN_KEY,
             MAX_TRIGGER_VALUE,
         );
-        if records_for_user.len() > MAX_QUERY_SIZE - total_count {
-            records_for_user.truncate(MAX_QUERY_SIZE - total_count);
+        if records_for_user.len() > QUERY_SIZE - total_count {
+            records_for_user.truncate(QUERY_SIZE - total_count);
         }
```
Comment (martinthomson) [inline benches/oneshot/ipa.rs:38]: You can just truncate unconditionally.

```suggestion
        records_for_user.truncate(QUERY_SIZE - total_count);
```

**Review by martinthomson — APPROVED** [review]
LGTM.  Check with @akoshelev about #562, but this one looks more ready to me.

**Review by benjaminsavage — APPROVED** [review]

**Pull request merged**